### PR TITLE
Generalize the CropClipper into the ProxyManagerClipper 

### PR DESCRIPTION
**Pull request by hexonaut**
It seems likely we will be using this Proxy / Manager paradigm moving forward. It's been used for `dss-crop-join`, but also being used for `dss-charter`. It makes sense to generalize the clipper if we can. This is my attempt to do so using callbacks to keep the state in a good place. It's maybe not the most elegant solution, and it seems like we may run into edge cases in the future, but it should work for both examples so far.

Luckily both `onLiquidation` and `onVatFlux` can be done in a permissionless way, but if in the future these need to be permissioned we can do so by authorizing the clipper on the manager. This doesn't change the `ProxyManagerClipper`, so I think it's okay to permission the callbacks if needed.

### CODE REVIEW

File: src/ProxyManagerClipper.sol [hunk 1]
```diff
@@ -406,15 +410,16 @@ contract CropClipper {
...
             // Send collateral to the UrnProxy of who
             {
-                address urp = cropMgr.proxy(who);
-                require(urp != address(0), "CropClipper/no-urn-proxy");
-                vat.flux(ilk, address(this), urp, slice);
-                crop.tack(address(this), urp, slice);  // Transfer stake+rewards
+                address urp = proxyMgr.proxy(who);
+                require(urp != address(0), "ProxyManagerClipper/no-urn-proxy");
+                vat.flux(ilk, address(this), proxy, slice);
+                proxyMgr.onVatFlux(address(gemJoin), address(this), proxy, slice);
```
Comment (kmbarry1) [inline src/ProxyManagerClipper.sol:416]: We might as well just transfer to the Clipper's proxy in `kick` IMO.

**Comment by kmbarry1** [comment]
This gets the job done and prevents inheritance--unless we come up with something better, we should just go with this.

**Review by gbalabasquer — APPROVED** [review]
LGTM. I've just merged from master so it can be merged now to it.
The only question I'd have is if we should extract this contract to a different repo (maybe after merging this).

**Comment by talbaneth** [comment]
@hexonaut are we sticking with leaving the proxyManagerClipper in this repo?
If so I will add it (dss-crop-join) as a submodule in dss-charter.

**Pull request merged**